add git annex

notestaff opened the pull request:
This PR adds only a Linux recipe for git-annex.  That recipe works fine (the full git-annex test suite passes).  The MacOS recipe is proving more difficult.  I'll keep working on that, but in the meantime it'd be useful to have git-annex in conda-forge at least for Linux.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/git-annex```, ```recipes/rsync/rsync```) and found some lint.

Here's what I've got...


For **recipes/git-annex**:

 * There are 1 too many lines.  There should be one empty line at the end of the file.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/git-annex```, ```recipes/popt```) and found some lint.

Here's what I've got...


For **recipes/git-annex**:

 * There are 1 too many lines.  There should be one empty line at the end of the file.

For **recipes/popt**:

 * The top level meta keys are in an unexpected order. Expecting ['package', 'source', 'build', 'test', 'about'].
 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/git-annex```, ```recipes/popt```, ```recipes/rsync```) and found some lint.

Here's what I've got...


For **recipes/git-annex**:

 * There are 1 too many lines.  There should be one empty line at the end of the file.

For **recipes/popt**:

 * The top level meta keys are in an unexpected order. Expecting ['package', 'source', 'build', 'test', 'about'].
 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/git-annex```, ```recipes/popt```, ```recipes/rsync```) and found it was in an excellent condition.

notestaff commented:
getting git-annex into conda-forge would help with PR #1720

notestaff commented:
this PR also includes recipes for rsync and popt, same as PR #4150 , as these dependencies of git-annex are not yet in conda-forge.

yarikoptic commented:
woohoo!
Kudos @notestaff for this work!
If needed, I bet we could ask @joeyh (git-annex author) to advice on particular aspects of building for OSX (and Windows... he was fixing up Windows build recently)

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/git-annex```) and found it was in an excellent condition.

notestaff commented:
@conda-forge/core Would you consider merging this without a working osx build, or is that a requirement?

notestaff commented:
No, this is using ghc installed by Stack .

jakirkham commented:
Building on Linux alone is fine. Would raise an issue on the feedstock for adding macOS.

isuruf commented on recipes/git-annex/build.sh:
```diff
@@ -0,0 +1,31 @@
+export C_INCLUDE_PATH=${PREFIX}/include
+export CPLUS_INCLUDE_PATH=${PREFIX}/include
+export LDFLAGS=-L${PREFIX}/lib
+export LIBRARY_PATH=${PREFIX}/lib
```
Do you need the env variables above? `toolchain` sets some of these.

notestaff replied:
I've tried removing some of these, got errors -- see earlier commits.

isuruf replied:
Then you should append instead of setting the env variables

notestaff replied:
@isuruf this has been fixed.

yarikoptic commented:
FWIW, if this one is merged, it would remove the road-block toward #1720 as well (if only on Linux to start with)

notestaff commented:
@isuruf I removed all but one setting of environment variables in build.sh , and made that one (LIBRARY_PATH) append to any existing path.  Also updated to latest version of git-annex.  Let me know if you see other issues.

luispedro commented:
Building Haskell packages on Mac OS X is an issue that bioconda also faces:

https://github.com/bioconda/bioconda-recipes/issues/6607

I would imagine that whenever someone finds a robust solution, it'd apply to all of these packages.

luispedro commented:
Also, +1 on adding git-annex: I would love to be able to install it with `conda`.

luispedro commented:
Any reason this hasn't been merged?

yarikoptic commented:
I wish i could do that ;-)

dougalsutherland commented:
Sorry for the slow movement on this one, especially after you did so much work @notestaff.

I just:
- rebased onto `master`, since the CI scripts changed in the meantime
- updated some pinnings to [the preferred versions](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml)
- updated to the latest version.

Assuming Circle passes, it then looks good to me.

notestaff commented:
@dougalsutherland thanks a lot!

dougalsutherland commented:
Merged! Thanks for all your work on this @notestaff.

yarikoptic commented:
awesome -- thank you @notestaff and @dougalsutherland for making it happen!
